File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> indxToDiffMap;
        vector<int> res;
        for(int i = 0; i < nums.size(); i++){
            int diff = target - nums[i];
            if(indxToDiffMap.contains(diff)){
                vector<int> res = {indxToDiffMap[diff], i};
                return res;
            }
            indxToDiffMap[nums[i]] = i;
        }

        return res;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r = 0,1

        res = 0
        while r < len(prices):
            res = max(prices[r] - prices[l], res)
            if prices[l] > prices[r]:
                l=r
                r+=1
            else:
                r += 1
            
        return res